nistats/glm.py

```diff
@@ -58,183 +47,394 @@ def data_scaling(Y):
+                'contrasts must be a sequence of %d session contrasts' %
+                len(self.results_))
+
+        contrast = None
+        for i, (labels_, results_, con_val) in enumerate(zip(
+                self.labels_, self.results_, con_vals)):
+            if np.all(con_val == 0):
+                warn('Contrast for session %d is null' % i)
+            contrast_ = compute_contrast(labels_, results_, con_val,
+                                         contrast_type)
+            if contrast is None:
+                contrast = contrast_
             else:
-                contrast_type = 'F'
-        if contrast_type not in ['t', 'F', 'tmin-conjunction']:
-            raise ValueError('Unknown contrast type: %s' % contrast_type)
-
-        effect_ = np.zeros((dim, self.labels_.size), dtype=np.float)
-        var_ = np.zeros((dim, dim, self.labels_.size), dtype=np.float)
-        if contrast_type == 't':
-            for l in self.results_.keys():
-                resl = self.results_[l].Tcontrast(con_val)
-                effect_[:, self.labels_ == l] = resl.effect.T
-                var_[:, :, self.labels_ == l] = (resl.sd ** 2).T
-        else:
-            for l in self.results_.keys():
-                resl = self.results_[l].Fcontrast(con_val)
-                effect_[:, self.labels_ == l] = resl.effect
-                var_[:, :, self.labels_ == l] = resl.covariance
+                contrast = contrast + contrast_
 
-        dof_ = self.results_[l].df_resid
-        return Contrast(effect=effect_, variance=var_, dof=dof_,
-                        contrast_type=contrast_type)
+        if output_z or output_stat:
+            # compute the contrast and stat
+            contrast.z_score()
+
+        # Prepare the returned images
+        do_outputs = [output_z, output_stat, output_effects, output_variance]
+        estimates = ['z_score_', 'stat_', 'effect', 'variance']
+        descrips = ['z statistic', 'Statistical value', 'Estimated effect',
+                    'Estimated variance']
+        output_images = []
+        for do_output, estimate, descrip in zip(
+                do_outputs, estimates, descrips):
+            if not do_output:
+                continue
+            estimate_ = getattr(contrast, estimate)
+            if estimate_.ndim == 3:
+                shape_ = estimate_.shape
+                estimate_ = np.reshape(estimate_,
+                                       (shape_[0] * shape_[1], shape_[2]))
+            output = self.masker_.inverse_transform(estimate_)
+            output.get_header()['descrip'] = (
+                '%s of contrast %s' % (descrip, contrast_name))
+            output_images.append(output)
+        return output_images
+
+    def fit_transform(
+        self, design_matrices, fmri_images, con_vals, contrast_type=None,
+        contrast_name='', output_z=True, output_stat=False,
+        output_effects=False, output_variance=False):
+        """ Fit then transform"""
```

Comment:
Maybe add something like "see FirstLevelGLM.fit and FirstLevelGLM.transform documentation for more details."